jni/src/main/java/com/horizen/cswnative/CswProof.java

```diff
@@ -0,0 +1,342 @@
+package com.horizen.cswnative;
+
+import java.util.Optional;
+
+import com.horizen.certnative.WithdrawalCertificate;
+import com.horizen.librustsidechains.FieldElement;
+import com.horizen.provingsystemnative.CreateProofResult;
+import com.horizen.provingsystemnative.ProvingSystemType;
+
+public class CswProof {
+    private static native boolean nativeSetup(
+        ProvingSystemType psType,
+        int rangeSize,
+        int mstTreeHeight,
+        int ftTreeHeight,
+        int scTxCommTreeHeight,
+        String provingKeyPath,
+        String verificationKeyPath,
+        boolean zk,
+        int maxProofPlusVkSize,
+        boolean compressPk,
+        boolean compressVk
+    );
+
+    /**
+     * Generate (provingKey, verificationKey) pair for this circuit.
+     * @param psType - proving system to be used
+     * @param rangeSize - number of blocks between `mcbScTxsComStart` and `mcbScTxsComEnd`
+     * @param mstTreeHeight - Height of the MST, i.e. the SC Utxo tree
+     * @param ftTreeHeight - Height of the FT subtree of the Commitment Tree of a particular SC
+     * @param scTxCommTreeHeight - Height of the subtree of the SCsCommitmentTree having as leaves the single ScTxsCommitments 
+     * @param provingKeyPath - file path to which saving the proving key
+     * @param verificationKeyPath - file path to which saving the verification key
+     * @param maxProofPlusVkSize - maximum allowed size for proof + vk
+     * @param compressPk - if the proving key must be saved to provingKeyPath in compressed form
+     * @param compressVk - if the verification key must be saved to verificationKeyPath in compressed form
+     * @return true if (pk, vk) generation and saving to file was successfull, false otherwise
+     */
+    public static boolean setup(
+        ProvingSystemType psType,
+        int rangeSize,
+        int mstTreeHeight,
+        int ftTreeHeight,
+        int scTxCommTreeHeight,
```

Comment:
why not to hardcode `scTxCommTreeHeight` and `ftTreeHeight` on rust side getting the values from CCTP commitment tree?

Reply:
You're right, it was just to be more generic, but maybe it's not needed. These values are specified in the Constants class anyway.